### PR TITLE
Pin golangci-lint and buf versions

### PR DESCRIPTION
**Pull request by codingllama**
Our "latest" experiments weren't happy ones, so pinning the versions again.

### CODE REVIEW

**Comment by codingllama** [comment]
If the devbox finishes building I'll pin it here too.

**Comment by codingllama** [comment]
Devbox pinned.

**Review by zmb3 — APPROVED** [review]

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -309,11 +309,14 @@ RUN go install github.com/daixiang0/gci@v0.11.0
 RUN go install gotest.tools/gotestsum@v1.10.1
 
 # Install golangci-lint.
-RUN curl -fsSL "https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh" | \
-    sh -s -- -b "$(go env GOPATH)/bin"
+RUN VERSION='v1.54.2'; \
```
Comment (jakule) [inline build.assets/Dockerfile:312]: Should we move this and buf version to `versions.mk` to keep all versions in one place? I think it should also be possible to use that in GHA.

**Review by jakule — APPROVED** [review]

File: build.assets/Dockerfile (hunk 1 above)
Comment (codingllama) [reply build.assets/Dockerfile:312]: BUF_VERSION is there. I don't see how I can cleanly use the version from versions.mk in GHA, if you know the right incantation please share it.

**Pull request merged**